src/initialize_workspace.py

```diff
@@ -0,0 +1,95 @@
+import json,adal,requests
+from requests.exceptions import HTTPError
+import logging
+
+from msrestazure.azure_active_directory import AADTokenCredentials
+
+log = logging.getLogger()
+
+DB_RESOURCE_ID = '/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Databricks/workspaces/{}'
+LIST_NODE_TYPES_ENDPOINT = '/api/2.0/clusters/list-node-types'
+
+# AAD access token is to authentication what management token is to authorization.
+
+def adb_authenticate_client_key(params):
+	"""
+    Authenticate using service principal w/ key.
+    """
+	authority_host_uri = 'https://login.microsoftonline.com'
+	tenant = params['tenant_id']
+	authority_uri = authority_host_uri + '/' + tenant
+	resource_uri = params['databricks_resource_id']
+	client_id = params['client_id']
+	client_secret = params['client_secret']
+	context = adal.AuthenticationContext(authority_uri, api_version=None)
+	mgmt_token = context.acquire_token_with_client_credentials(resource_uri, client_id, client_secret)
+	credentials = AADTokenCredentials(mgmt_token, client_id)
+	log.info("created aad authentication token")
+	return credentials
+
+
+'''
+The management token is used for the admin login flows (users/service principal with Contributor access)
+- If the user/sp does not exist in the workspace, they will be automatically created as an admin user
+Using the bearer token (AAD access token) we cannot query the role or permission of the user/sp in the Azure Databricks Resource, 
+but using the management token we are able to query AAD for the role of the user/sp
+'''
+
+
+def adb_authorization_client_key(params):
+	"""
+    Authenticate using service principal w/ key.
+    """
+	authority_host_uri = 'https://login.microsoftonline.com'
+	tenant = params['tenant_id']
+	authority_uri = authority_host_uri + '/' + tenant
+	resource_uri = 'https://management.core.windows.net/'
```

Comment:
Change as global (in case we need to parameterize later)

Reply:
done